mod/redir.php

```diff
@@ -151,7 +151,7 @@ function redir_magic($a, $cid, $url)
 	// We don't use magic auth when there is no visitor, we are on the same system or we visit our own stuff
 	if (empty($visitor) || Strings::compareLink($basepath, System::baseUrl()) || Strings::compareLink($contact_url, $visitor)) {
 		Logger::info('Redirecting without magic', ['target' => $target_url, 'visitor' => $visitor, 'contact' => $contact_url]);
-		$a->redirect($target_url);
+		System::externalRedirect($target_url);
```

Comment:
Why not `DI::baseUrl()->redirect();` here?

Reply:
All `DI::baseUrl()->redirect();` does is prepending the base URL to a relative path and then calling `System::externalRedirect`. If we know we already have an external URL we can skip the BaseURL part.

Reply:
I just checked the function. I thought that we had a check that the path had to be external only. But this isn't the case (anymore). So it is okay.